src/components/BorrowerProfile/RepaymentSchedule.vue

```diff
@@ -217,23 +253,46 @@ export default {
 						return runningTotal + parseFloat(amount);
 					}, 0);
 
-					// format the repaymentDate & totalMonthlyPayment value
+					// push the monthly repayments to a new array to be used to check if repayments are on time
+					monthlyTotalRepayments.push(totalMonthlyPayment);
+
+					// iterate through the monthlyRepayments and add together each
+					// month's payments to get a total monthly payment amount
+					const totalMonthlyPaymentValue = monthlyTotalRepayments.reduce((runningTotal, monthlyAmount) => {
+						return runningTotal + parseFloat(monthlyAmount);
+					}, 0);
+
+					// set the repaid boolean
+					repaid = this.repaidAmount > totalMonthlyPaymentValue;
+
+					const now = new Date();
+					const parsedRepaymentDate = parseISO(repaymentDate);
+					// if a payment is not repaid and the repayment data is before now, then mark the loan delinquent
+					delinquent = !repaid && isBefore(parsedRepaymentDate, now);
```

Comment:
Seems like this should prevent the "Future" dates from being shown as delinquent but that doesn't appear to be the case in the attached screenshot which shows delinquency for several months into the future. Please see if you can get those future date to reflect the "expected" state instead of delinquent.

Reply:
@mcstover After looking into this a little, I think it was a result of how if forced delinquent repayments to show. In order to do that I changed line 271

delinquent = !repaid && isBefore(parsedRepaymentDate, now);
I updated that to: 
delinquent = !repaid && !isBefore(parsedRepaymentDate, now);

Doing this caused all the rows to show as delinquent. With real data, I don't think that will be an issue.